category_encoders/target_encoder.py

```diff
@@ -134,14 +155,19 @@ def _check_dict_key_tuples(self, d):
         return min_tuple_size == max_tuple_size, min_tuple_size
 
     def _fit(self, X, y, **kwargs):
-        if self.hierarchy:
+        if isinstance(self.hierarchy, dict):
+            if not set(self.cols).issubset(X.columns):
```

Comment:
this check is not specific to hierarchy right? if so it should be moved  to the base encoder. Strange that we do not have such a check implemented already...

Reply:
Agreed.  We have moved this, hopefully to the right place.